combat/CombatTargetting.cpp

```diff
@@ -0,0 +1,150 @@
+#include "CombatTargetting.h"
+
+#include "../universe/Condition.h"
+#include "../universe/Enums.h"
+#include "../universe/Planet.h"
+#include "../universe/Fleet.h"
+#include "../universe/Ship.h"
+#include "../universe/ShipDesign.h"
+#include "../universe/Fighter.h"
+#include "../universe/ValueRef.h"
+
+#include "../util/Logger.h"
+#include "../util/Random.h"
+
+bool Targetting::isPreferredTarget(Targetting::TriggerConditions conditions,
+                                   Targetting::Target target) {
+    const Precision highest_weight = Targetting::findPrecision(conditions);
+    for (long unsigned int i = 0; i < conditions.conditions.size(); i++) {
```

Comment:
prefer `for (const auto& condition : conditions.conditions) {` unless you need the `i` index specifically...

Reply:
i need it for the conditions.weights[i] - maybe i get round to refactor that structure, but dunno